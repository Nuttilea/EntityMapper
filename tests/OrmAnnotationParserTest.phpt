<?php

namespace Test;

use Nutillea\OrmAnotationPareser;
use Nette;
use Tester;
use Tester\Assert;

$container = require __DIR__ . '/bootstrap.php';


/**
 * @testCase
 *
 * @property int $id orm:column(ID)
 * @property int $tag orm:column
 * @property string $name orm:belongsTo(user_name)
 *
 */
class OrmAnnotationParserTest extends Tester\TestCase
{
	private $container;


	public function __construct(Nette\DI\Container $container)
	{
		$this->container = $container;
	}


	public function setUp(){

    }



	public function testSimpleAnnotations()
	{
        $annotations = Nette\Reflection\AnnotationsParser::getAll(new \ReflectionClass(self::class));
        $r = OrmAnotationPareser::parseOrmPropertiesTags($annotations['property']);
        Assert::equal([
            'id' => [
                'column' => 'ID',
                'type' => 'int'
            ],
            'tag' => [
                'column' => true,
                'type' => 'int'
            ],
            'name' => [
                'belongsTo'=>'user_name',
                'type' => 'string'
            ]
        ], $r);

    }
}


$test = new OrmAnnotationParserTest($container);
$test->run();
