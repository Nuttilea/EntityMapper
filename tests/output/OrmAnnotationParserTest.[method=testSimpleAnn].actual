[
	'id' => ['type' => 'int ', 'column' => 'ID'],
	'tag' => ['type' => 'int ', 'column' => true],
	'name' => ['type' => 'string ', 'belongsTo' => 'user_name'],
]
