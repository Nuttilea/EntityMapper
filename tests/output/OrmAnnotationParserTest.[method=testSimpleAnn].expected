[
	'id' => ['column' => 'ID', 'type' => 'int'],
	'tag' => ['column' => true, 'type' => 'int'],
	'name' => ['belongsTo' => 'user_name', 'type' => 'string'],
]
